slider/beatmap.py

```diff
@@ -1680,6 +1683,31 @@ def _resolve_stacking_old(self, hit_objects, ar, cs):
 
         return hit_objects
 
+    def closest_hitobject(self, t):
+        """The hitobject with a ``time`` closest to ``t`` (in ms).
+        """
+        if not self._hit_object_times:
+            self._hit_object_times = [hitobj.time.total_seconds() * 1000
+                                      for hitobj in self._hit_objects]
+        i = np.searchsorted(self._hit_object_times, t)
+        # if ``t`` is after the last hitobject, an index of
+        # len(self._hit_objects) will be returned. The last hitobject will
+        # always be the closest hitobject in this case.
+        if i == len(self._hit_objects):
+            return self._hit_objects[-1]
+        # similar logic follows for the first hitobject.
+        if i == 0:
+            return self._hit_objects[0]
+
+        # searchsorted tells us the two closest hitobjects, but not which is
+        # closer. Check both candidates.
+        hitobj1 = self._hit_objects[i - 1]
```

Comment:
This would be dumb, but we may want to also guard against a beatmap with 1 hit object, like maybe some spinner practice map. I am not sure it is worth checking explicitly for a 0 hit object map.